### ISSUE TITLE
Cargo test with `--concurrent` fails

### ISSUE THREAD

**Issue by vlopes11** [0xPolygonMiden/miden-vm#437]
If a test with all features is executed, it fails for the following cases

```
⚡ cargo test --all-features
(...)
test examples::fibonacci::test_fib_example ... FAILED
test examples::fibonacci::test_fib_example_fail ... FAILED
```

We should consider adding two new steps to the CI: one with `--all-features` & other with `--no-default-features`

**Comment by vlopes11** [comment]
Update: the feature that is broken is `concurrent`

**Comment by vlopes11** [comment]
Fixed by #708
